src/realm/object-store/object_changeset.cpp

```diff
@@ -27,10 +27,12 @@ void ObjectChangeSet::insertions_add(ObjectKeyType obj)
 
 void ObjectChangeSet::modifications_add(ObjectKeyType obj, ColKeyType col)
 {
-    // don't report modifications on new objects
-    if (m_insertions.find(obj) == m_insertions.end()) {
-        m_modifications[obj].insert(col);
+    // don't report modifications on new objects. treat objects that have been deleted and recreated
```

Comment:
This is something we specifically don't want to do; if the user calls delete and then create within a write transaction they should get a notification of a delete and a create.

Reply:
Thanks for your review! I'll figure that out then.

Reply:
It seems like something that inherently has to be fixed on the server if we're getting delete+create instructions from an update call, because we can't infer intent and figure out if it's supposed to be a delete+create or an update after the fact.

I think we also have a bug in core where delete+create is actually just sending no notification (https://github.com/realm/realm-core/pull/4167 https://jira.mongodb.org/browse/RCORE-368)? Fixing that's the immediate thing you could probably do in core.

Reply:
It doesn't seem obvious to me that a mongo replace should be translated to a delete+insert rather than just setting each of the fields. It doesn't behave like a deletion (e.g. it doesn't break incoming links) and I don't really see a reason why it'd need to emit a delete+insert when translating the operation to Realm?

Reply:
I also noticed this behavior when looking at some logs:
```
EraseObject     table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}\
CreateObject    table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="referenceId", path=[], value=String("000040079100"), default=0\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="name", path=[], value=String("40079100"), default=0\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="title", path=[], value=String("david test BD04 - assign user6"), default=0\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="hasQuota", path=[], value=Bool(0), default=0\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="nextRootNode", path=[], value=Link(target_table = ReportingTreeNode, target = ObjectId\{5fc628740d23f200067c4179\}), default=0\
Set             table="ReportingTreeNode", object=ObjectId\{5fc0ca2193ecee94a51ff84b\}, field="_partitionKey", path=[], value=String("SHARED"), default=0\
```
And the object didn't even exist before.